Update DoD to include new item about Istio operator changes (#625)
kyma-project/istio: .github/ISSUE_TEMPLATE/feature_request.md

- [ ] Provide documentation.
- [ ] Verify if the solution works for both open-source Kyma and SAP BTP, Kyma runtime.
- [ ] If you changed the resource limits, explain why it was needed.

> - [ ] If the default configuration of the Istio operator has been changed, a manual upgrade test was done to verify that the change can be rolled out correctly.

Reviewer: ```suggestion
- [ ] If the default configuration of Istio Operator has been changed, you performed a manual upgrade test to verify that the change can be rolled out correctly.
```
